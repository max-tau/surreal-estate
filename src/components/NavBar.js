import React from "react";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import "../styles/navbar.css";

const NavBar = ({ onLogin, userId, onLogout }) => {
  return (
    <div className="navbar">
      <img
        className="navbar-logo"
        src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
        alt="Surreal Estate logo"
      />
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link className="navbar-links-item" to="/">
            View Properties
          </Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-item" to="/add-property">
            Add a Property
          </Link>
        </li>
        {userId ? (
          <li className="navbar-links-item">
            <Link className="navbar-links-item" to="/favourites">
              View Favourites
            </Link>
          </li>
        ) : null}
      </ul>
      {userId ? (
        <button className="navbar-signout" type="submit" onClick={onLogout}>
          Sign out
        </button>
      ) : (
        <div className="navbar-login">
          <Link className="navbar-links-auth" to="/create-account">
            Create Account
          </Link>
          <Link className="navbar-links-auth" to="/sign-in">
            Sign In
          </Link>
        </div>
      )}
    </div>
  );
};

export default NavBar;
