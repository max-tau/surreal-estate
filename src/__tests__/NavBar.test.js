import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  const handleLogin = () => {
    jest.fn();
  };
  const handleLogout = () => {
    jest.fn();
  };

  it("renders component correctly", () => {
    const { asFragment } = render(
      <Router>
        <NavBar onLogin={handleLogin} userId={1} onLogout={handleLogout} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("contains an image with correct src", () => {
    render(
      <Router>
        <NavBar onLogin={handleLogin} userId={1} onLogout={handleLogout} />
      </Router>
    );
    const logo = screen.getByRole("img");

    expect(logo).toHaveAttribute(
      "src",
      "https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
    );
  });
});
