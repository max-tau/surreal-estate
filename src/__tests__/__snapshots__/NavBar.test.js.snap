// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders component correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="Surreal Estate logo"
      class="navbar-logo"
      src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
    />
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          class="navbar-links-item"
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          class="navbar-links-item"
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
    </ul>
    <button
      class="navbar-signout"
      type="submit"
    >
      Sign out
    </button>
  </div>
</DocumentFragment>
`;
